inMemory.go

```diff
@@ -24,3 +24,8 @@ func (r *inMemoryRepository) Save(device *Device) (*DeviceWithId, error) {
 	r.mu.Unlock()
 	return &deviceWithId, nil
 }
+
+func (r *inMemoryRepository) GetById(id int) (*DeviceWithId, error) {
+	device := r.devices[id]
+	return &device, nil
```

Comment:
This implementation will return an empty object when device was not found. I would change it to 

```suggestion
        if device, ok := r.devices[id]; ok {
           return &device, nil
         }
	return nil, nil
```